### DISCUSSION TITLE
BasicHDU.getBitpix() is now deprecated

### DISCUSSION THREAD

**Discussion by bogdanni** [nom-tam-fits/nom-tam-fits#440] [Q&A]
How can a program retrieve the FITS data type in order to process into Java data type accordingly?

**Comment by attipaci** [comment]
Sorry that was a mistake. I'll have it back undeprecated in the next rc (in a few days). I'm still working out some nagging issues...

On the other hand, the type is only relevant in image data , that is `ImageHDU` (or `RandomGroupsHDU`). These could provide a more Java friendly way to report it, e.g. a `getImageClass()` method to return the base type, such as `int.class` or `double.class`. BITPIX should probably be reserved for internal use. It has meaning only for the FITS data format but not on the Java side...

**Comment by bogdanni** [comment]
In the end I did it with `ImageHDU.getData().getData()` and a chain of `if instanceof`.
The API of the library is quite confusing, looking forward for the cleanup in version 2.

Thank you for the great work!

**Reply by attipaci** [reply]
I fully agree about the confusing API. (It drives me nuts at times). As I said, I'll leave `getBitpix()` for now. As for 2.0, I'm not sure when it will ever arrive...

**Reply by bogdanni** [reply]
At least you make it advance, it seemed abandoned for years, please keep it up.
